src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -806,3 +809,19 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
         }
     # endregion
 
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain the 10 status complete file but remove older .complete.status files """
+        list_of_files = glob.glob(status_folder + '/' + '*.complete.status')
+        files_to_removed = []
+
+        if len(list_of_files) <= Constants.MAX_COMPLETE_STATUS_FILES_COUNT:
+            return
+
+        while len(list_of_files) > Constants.MAX_COMPLETE_STATUS_FILES_COUNT:
+            oldest_file = min(list_of_files, key=lambda x: (os.path.getmtime(x), int(re.search(r'(\d+)\.complete.status', x).group(1)), x))
```

Comment:
you don't need to search the directory again to sort files based on timestamp. See line 59 in delete_older_log_files_per_module() in FileLogger.py under extension/src/local_loggers/. That function also retains certain number of files while deleting others

Reply:
done